src/darwin/Framework/CHIP/CHIPPersistentStorageDelegateBridge.mm

```diff
@@ -112,43 +114,50 @@
     return error;
 }
 
-void CHIPPersistentStorageDelegateBridge::AsyncSetKeyValue(const char * key, const char * value)
+CHIP_ERROR CHIPPersistentStorageDelegateBridge::SyncSetKeyValue(const char * key, const void * value, uint16_t size)
 {
+    std::string base64Value = StringToBase64(std::string(static_cast<const char *>(value), size));
+
     NSString * keyString = [NSString stringWithUTF8String:key];
-    NSString * valueString = [NSString stringWithUTF8String:value];
-    dispatch_async(mWorkQueue, ^{
```

Comment:
@sagar-apple or @shana-apple could you look at this, please?

Reply:
I am unsure why this got removed - will add it back.